qiskit/dagcircuit/dagcircuit.py

```diff
@@ -335,28 +334,27 @@ def apply_operation_back(self, op, qargs=None, cargs=None, condition=None):
 
         return node
 
-    def apply_operation_front(self, op, qargs, cargs, condition=None):
+    def apply_operation_front(self, op, qargs, cargs):
         """Apply an operation to the input of the circuit.
 
         Args:
             op (qiskit.circuit.Instruction): the operation associated with the DAG node
             qargs (list[Qubit]): qubits that op will be applied to
             cargs (list[Clbit]): cbits that op will be applied to
-            condition (tuple or None): optional condition (ClassicalRegister, value)
 
         Returns:
             DAGNode: the current max node
 
         Raises:
             DAGCircuitError: if initial nodes connected to multiple out edges
         """
-        all_cbits = self._bits_in_condition(condition)
+        all_cbits = self._bits_in_condition(op.condition)
         all_cbits.extend(cargs)
 
-        self._check_condition(op.name, condition)
+        self._check_condition(op.name, op.condition)
         self._check_bits(qargs, self.input_map)
         self._check_bits(all_cbits, self.input_map)
-        node = self._add_op_node(op, qargs, cargs, condition)
+        node = self._add_op_node(op, qargs, cargs, op.condition)
```

Comment:
It would be great if in this PR you could extent this fix to cover `_add_op_node` (which wouldn't need `condition` to be deprecated) and even the `DAGNode` constructor (which would).

Reply:
Thank you bringing this up! I wanted ask if I should've handled this in a separate PR or not. I'll add these changes to this PR then.